full-service/src/service/receipt.rs

```diff
@@ -185,29 +185,34 @@ where
         &self,
         address: &str,
         receiver_receipt: &ReceiverReceipt,
-    ) -> Result<(ReceiptTransactionStatus, Option<TxoDetails>), ReceiptServiceError> {
+    ) -> Result<(ReceiptTransactionStatus, Option<Txo>), ReceiptServiceError> {
         let conn = &self.wallet_db.get_conn()?;
         conn.transaction(|| {
             let assigned_address = AssignedSubaddress::get(address, &conn)?;
-            let account_id = AccountID(assigned_address.account_id_hex);
+            let account_id = AccountID(assigned_address.account_id_hex.clone());
             let account = Account::get(&account_id, &conn)?;
             // Get the transaction from the database, with status.
-            let txos_and_statuses =
-                Txo::select_by_public_key(&account_id, &[&receiver_receipt.public_key], &conn)?;
+            let txos = Txo::select_by_public_key(&[&receiver_receipt.public_key], &conn)?;
 
             // Return if the Txo from the receipt is not in this wallet yet.
-            if txos_and_statuses.is_empty() {
+            if txos.is_empty() {
                 return Ok((ReceiptTransactionStatus::TransactionPending, None));
             }
-            let (txo, status) = &txos_and_statuses[0];
+            let txo = txos[0].clone();
 
-            // Figure out whether the Txo was minted by us, and has not yet been received by
-            // us. (For to-self transactions). If the Txo was minted by us, this
-            // transaction is pending.
-            if status.txo_type == TXO_TYPE_MINTED && status.txo_status == TXO_STATUS_SECRETED {
-                return Ok((ReceiptTransactionStatus::TransactionPending, None));
+            // Return if the Txo from the receipt has a pending tombstone block index
+            if txo.pending_tombstone_block_index.is_some() {
+                return Ok((ReceiptTransactionStatus::TransactionPending, Some(txo)));
+            }
+
+            // We are reproducing a bug with this logic and it is not how the
```

Comment:
Can you please update the comment to explain the bug, current behavior and desired behavior?

Reply:
We are about to fix this in the next PR. We'll put a full writeup in that PR.